### PR TITLE
fix(ci): expand changelog gate to cover template and scaffolding paths

### LINKED ISSUE #1156: fix(ci): changelog gate misses scaffolding and template file changes - only guards src/
﻿## Problem

The changelog gate in `.github/workflows/squad-ci.yml` only triggers when files matching `^packages/squad-(sdk|cli)/src/` are changed (line 222). This means PRs that touch scaffolding, template, or agent files bypass the changeset requirement silently.

## What is missed

Any PR modifying these paths skips the changelog gate entirely:

- `.squad-templates/*.md` — Coordinator, Scribe charter, spawn-reference, all on-demand references
- `templates/skills/**/*.md` — Skill files bundled into the CLI
- `packages/squad-cli/templates/**/*.md` — CLI-bundled scaffolding
- `packages/squad-sdk/templates/**/*.md` — SDK-bundled scaffolding
- `.squad/agents/*/charter.md` — Agent charters shipped with the repo

Changes to these files are **user-facing behavioral changes** — they alter what agents do, how they spawn, what invariants they enforce, and what templates users get on `squad init`. They are every bit as impactful as TypeScript source changes.

## Real-world impact

- PR #1035 modified `.squad-templates/squad.agent.md`, `.squad-templates/scribe-charter.md`, and multiple `templates/skills/` files without a changeset — the gate passed silently because no `src/` files were touched.
- CONTRIBUTING.md states changesets are required for all PRs, but the CI gate only enforces this for `src/` changes, creating a gap between documented policy and automated enforcement.
- AI contributors (GitHub Copilot, Squad agents) follow CI signals — if the gate passes, they assume no changeset is needed.

## Proposed Fix

Expand the `SDK_CLI_CHANGED` pattern in `.github/workflows/squad-ci.yml` to also match scaffolding and template paths:

```bash
SDK_CLI_CHANGED=$(echo "$CHANGED" | grep -E \
  '^packages/squad-(sdk|cli)/src/|^packages/squad-(sdk|cli)/templates/|^\.squad-templates/|^templates/' \
  || true)
```

This ensures any PR that ships behavioral changes to bundled templates or agent scaffolding must include a changeset, consistent with CONTRIBUTING.md intent.

## Introduced by / exacerbated by

- `.github/workflows/squad-ci.yml` line 222 — pattern too narrow
- PR #1035 merged without a changeset (retroactively added in PR #1154)

### PR DESCRIPTION
**Pull request by obit91**
﻿## Summary

The `changelog-gate` job in `.github/workflows/squad-ci.yml` computed `SDK_CLI_CHANGED` using a pattern that only matched `^packages/squad-(sdk|cli)/src/`. As a result, PRs that touched bundled templates, scaffolding, or agent files bypassed the changeset requirement silently.

This change expands the `SDK_CLI_CHANGED` grep pattern to also match:
- `packages/squad-(sdk|cli)/templates/`
- `.squad-templates/`
- `templates/`

The two related summary/echo strings were updated to accurately reflect that the gate now also covers template changes ("no SDK/CLI source or template changes" / "SDK/CLI source or template files changed:"). No other logic was changed.

Closes #1156

## Scope

- **repo-health** change ΓÇö touches only `.github/workflows/squad-ci.yml`.
- No product source modified (`packages/*/src/` untouched).
- No changeset required (does not modify `packages/squad-sdk/src` or `packages/squad-cli/src`).

## Testing

- Sanity-tested the new `grep -E` pattern against sample paths: `packages/squad-sdk/src/z.ts`, `packages/squad-cli/templates/y.md`, `.squad-templates/foo.md`, and `templates/skills/x/SKILL.md` all **match**, while `README.md` does **not** match.
- Validated the workflow YAML remains well-formed (`js-yaml` parse ΓåÆ `YAML OK`).
- `npm run build` passes from the repo root (YAML-only change, build unaffected).

## Real-world example: PR #1035

[#1035](https://github.com/bradygaster/squad/pull/1035) ("fix: context overflow sentinel and coordinator size reduction", merged 2026-05-22) is a concrete case of this gap:

- It changed **30+ template/scaffolding files** across `packages/squad-cli/templates/`, `packages/squad-sdk/templates/`, `.squad-templates/`, and `templates/` (~2,000+ lines), plus `packages/squad-cli/package.json`.
- It touched **zero** `packages/squad-(sdk|cli)/src/` files.
- Under the old `src/`-only gate, `SDK_CLI_CHANGED` matched nothing -> the Changelog Gate reported **"Not applicable"** and passed -> **no changeset was required or added**, even though substantial shipped template content changed.
- With this PR's widened gate, a change like #1035 would correctly require a changeset (or `CHANGELOG.md` update, or the `skip-changelog` label).

## Automated gate test (added in this PR)

Added `test/ci/changelog-gate.test.ts` (vitest) to lock in the widened gate and guard #1156 against regression.

- Reads the **live** `SDK_CLI_PATH_REGEX` directly from `.github/workflows/squad-ci.yml` (single source of truth), builds a JS `RegExp`, and asserts a table of paths. If the regex variable is renamed/removed, the suite fails loudly.
- **8 governed paths must match** (gate requires a changeset): `packages/squad-sdk/src/`, `packages/squad-cli/src/`, both `packages/squad-(sdk|cli)/templates/` trees, `.squad-templates/`, and top-level `templates/`.
- **7 unrelated paths must not match** (gate stays silent): `README.md`, `docs/...`, `.github/workflows/squad-ci.yml`, `package.json`, `packages/squad-cli/package.json` (boundary case — not under `src/`/`templates/`), `scripts/...`, and the test file itself.
- Runs under `npm test` (`vitest run`, which only collects `test/**/*.test.ts`). `npx vitest run test/ci/changelog-gate.test.ts` -> **15 passed**; `npm run build` passes.

### CODE REVIEW

File: .github/workflows/squad-ci.yml [hunk 1]
```diff
@@ -219,12 +219,12 @@ jobs:
           BASE="${{ github.event.pull_request.base.sha }}"
           HEAD="${{ github.event.pull_request.head.sha }}"
           CHANGED=$(git diff --name-only "$BASE"..."$HEAD")
-          SDK_CLI_CHANGED=$(echo "$CHANGED" | grep -E '^packages/squad-(sdk|cli)/src/' || true)
+          SDK_CLI_CHANGED=$(echo "$CHANGED" | grep -E '^packages/squad-(sdk|cli)/src/|^packages/squad-(sdk|cli)/templates/|^\.squad-templates/|^templates/' || true)
```
[the remark this answers is not in the document]
Comment (obit91) [reply .github/workflows/squad-ci.yml]: Good call — consolidated the alternation into a single named `SDK_CLI_PATH_REGEX` variable in `80ff7bd`, so the pattern lives in one place and is safer to extend. The regex content is byte-for-byte identical; only the structure changed (no behavioral change).

[the remark this answers is not in the document]
Comment (obit91) [reply .github/workflows/squad-ci.yml]: Thanks — checked this carefully. In this repo the top-level `templates/` directory is **not** unrelated content: it is a full canonical mirror of `.squad-templates/`, generated by `scripts/sync-templates.mjs` (its `MIRROR_TARGETS` are `templates/`, `packages/squad-cli/templates/`, and `packages/squad-sdk/templates/`). So the entire tree is governed squad scaffolding that ships to users, and issue #1156 explicitly lists `templates/skills/**/*.md` among the paths that must be gated. Narrowing the match would re-open the exact gap this PR closes. I kept the path as-is and added an inline comment in `80ff7bd` documenting why the whole tree is intentionally governed.

**Comment by obit91** [comment]
### ✅ Added a regression test for the changelog gate

Per review request, this PR now includes an automated suite that verifies the gate actually catches the paths it's supposed to.

**New file:** `test/ci/changelog-gate.test.ts` (vitest)

**How it works:** the test reads the **live** `SDK_CLI_PATH_REGEX` from `.github/workflows/squad-ci.yml` (so it can't drift from the real gate), compiles it to a JS `RegExp`, and asserts a table of cases:

| Class | Examples | Expected |
|---|---|---|
| Governed (8 cases) | `packages/squad-sdk/src/index.ts`, `packages/squad-cli/src/cli/core/bootstrap.ts`, `packages/squad-{sdk,cli}/templates/...`, `.squad-templates/...`, `templates/...` | **match -> changeset required** |
| Unrelated (7 cases) | `README.md`, `docs/...`, `.github/workflows/squad-ci.yml`, `package.json`, `packages/squad-cli/package.json`, `scripts/...`, the test file | **no match -> not gated** |

If anyone renames/removes the regex variable or narrows it (re-opening the #1156 gap), the suite fails. Runs under `npm test`.

`npx vitest run test/ci/changelog-gate.test.ts` -> **15 passed**; `npm run build` passes.

Commit: bd3c9ac1

**Comment by obit91** [comment]
Added a changeset documenting this PR's changelog-gate scope widening. It records a patch bump for `@bradygaster/squad-cli` because template/scaffolding changes were previously able to ship without release-note entries; the widened gate ensures those user-facing template updates are no longer omitted.

**Comment by obit91** [comment]
cc @tamirdresher @bradygaster — heads up: this PR changes **how the changelog/changeset gate behaves**, which affects what lands in release notes going forward.

**What changes:**
- The `Changelog gate` in `.github/workflows/squad-ci.yml` previously required a changeset only for `packages/squad-(sdk|cli)/src/` changes. It now also covers template/scaffolding paths: `packages/squad-(sdk|cli)/templates/`, `.squad-templates/`, and top-level `templates/`.
- Net effect: user-facing template/scaffolding changes will now **require a changeset** and therefore show up in release notes, instead of being silently omitted.
- Added a regression test (`test/ci/changelog-gate.test.ts`) and a changeset documenting the change itself.

Flagging since this alters contributor workflow (more PRs will be asked for a changeset) and the release-note pipeline. Closes #1156.

**Comment by tamirdresher** [comment]
Good use of #1035 as the regression example — that kind of real-world evidence makes the case much cleaner than a hypothetical. The test-reads-live-regex approach is the right call.

A few things before this lands.

**`.squad/agents/*/charter.md` isn't in the regex.**

Issue #1156 explicitly called that path out as a bypass path. The current `SDK_CLI_PATH_REGEX` doesn't match `.squad/agents/troi/charter.md` or anything else under `.squad/agents/`. So strictly speaking, #1156 isn't fully closed by this PR. Either add `|^\.squad/agents/.+/charter\.md$` to the regex here, or open a follow-up issue and link it from #1156 — just don't close the issue until one of those happens.

**The "full canonical mirror" comment is wrong.**

The diff says: *"The top-level `templates/` tree is a full canonical mirror of `.squad-templates/` (see scripts/sync-templates.mjs)"*. It isn't. `sync-templates.mjs` copies from `.squad-templates/` into `templates/` but never deletes, so `templates/` is an append-only superset. Right now it has four files that have no counterpart in `.squad-templates/`: `ghost-protocol.md`, `loop.md`, `personal-charter.md`, and `skills/rework-rate/SKILL.md`. The gating is still correct — all four are governed scaffolding and should be behind the gate — but that comment is going to give the next reader a wrong mental model. Something like *"primary sync target of `.squad-templates/` — entire tree is governed scaffolding"* would be accurate.

**CONTRIBUTING.md is now out of date.**

Lines 190 and 217 both still say the changeset gate applies only to `packages/squad-(sdk|cli)/src/`. Worth updating in this PR so the docs and the actual gate match, otherwise contributors are going to wonder why their template-only change is asking for a changeset.

**One test hygiene note.**

In `changelog-gate.test.ts`, line 17 calls `expect(patternMatch, ...).not.toBeNull()` at module scope — outside any `describe` or `it` block. That assertion runs during vitest''s collection phase. If `SDK_CLI_PATH_REGEX` ever gets renamed in the YAML, the failure surfaces as "failed to collect tests" instead of a readable assertion error. Two-line fix: wrap it in `beforeAll`. Not a blocker, but it''s the kind of pattern that gets copied into other CI tests.

Fix the first three; the fourth is up to you but I''d do it. Core logic is sound, changeset is appropriate.

**Comment by obit91** [comment]
Thanks @tamirdresher — all addressed in 831cea8d.

1. **`.squad/agents/*/charter.md` now gated.** Added `|^\.squad/agents/.+/charter\.md$` to `SDK_CLI_PATH_REGEX`, so the last #1156 bypass path is covered. Extended the regression suite with a governed case (`.squad/agents/troi/charter.md` → matches) plus unrelated negatives (`.squad/agents/troi/notes.md`, `.squad/routing.md` → don't match). #1156 is now fully closed.
2. **Fixed the misleading comment.** Reworded to describe `templates/` as the *primary sync target of `.squad-templates/` — an append-only superset whose entire tree is governed scaffolding*, instead of "full canonical mirror".
3. **CONTRIBUTING.md updated.** Both the contributor handoff note (~L190) and the PR Readiness Checklist row (~L217) now list the template/scaffolding paths (`packages/squad-(sdk|cli)/templates/`, `.squad-templates/`, top-level `templates/`, `.squad/agents/*/charter.md`) alongside `src/`, matching the actual gate.
4. **Test hygiene.** Moved the module-scope `expect(...).not.toBeNull()` into `beforeAll` so a renamed/removed regex surfaces as a readable assertion rather than a "failed to collect tests" error.

Also refreshed the changeset to mention the charter path. `npx vitest run test/ci/changelog-gate.test.ts` → 18 passed; YAML validated; build green.

**Review by bradygaster — APPROVED** [review]
